rmw_fastrtps_cpp/src/ros_message_serialization.cpp

```diff
@@ -20,15 +20,20 @@
 bool
 _serialize_ros_message(
   const void * ros_message,
+  eprosima::fastcdr::FastBuffer & buffer,
   eprosima::fastcdr::Cdr & ser,
   void * untyped_typesupport,
   const char * typesupport_identifier)
 {
   if (using_introspection_c_typesupport(typesupport_identifier)) {
     auto typed_typesupport = static_cast<MessageTypeSupport_c *>(untyped_typesupport);
+    size_t estimated_size = typed_typesupport->getEstimatedSerializedSize(ros_message);
```

Comment:
what is the difference between `getEstimatedSerializedSize` and `ser.getSerializedDataLength` ?
I mean, the latter returns the exact size of all the serialized payload.
Is the estimated one supposed to be faster?

Reply:
The main point is knowing the size before serializing the data, so we can allocate the correct buffer size. For full bounded messages, the size is precalculated when creating the TypeSupport structure.